src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineTransportFactory.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Symfony\Bridge\Doctrine\RegistryInterface;
+use Symfony\Component\Messenger\Transport\Serialization\Serializer;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class DoctrineTransportFactory implements TransportFactoryInterface
+{
+    private $registry;
+    private $serializer;
+    private $debug;
+
+    public function __construct(RegistryInterface $registry, Serializer $serializer = null, $debug = false)
+    {
+        $this->registry = $registry;
+        $this->serializer = $serializer ?? Serializer::create();
+        $this->debug = $debug;
+    }
+
+    public function createTransport(string $dsn, array $options): TransportInterface
+    {
+        $manager = str_replace('doctrine://', '', $dsn);
+
+        $connection = new Connection($options['table_name'] ?? 'messenger_messages', $this->registry->getConnection($manager), $this->debug);
```

Comment:
I think it would be nice for DX to catch the `InvalidArgumentException` from `$this->registry->getConnection($manager)` and rethrow with some more context.

```
try {
    $connection = new Connection($options['table_name'] ?? 'messenger_messages', $this->registry->getConnection($manager), $this->debug);
} catch (\InvalidArgumentException $e) {
    throw new \RuntimeException(
        sprintf('Unable to create Doctrine transport for Messenger (dsn "%s"): %s', $dsn, $e->getMessage()),
        0,
        $e
    );
}

Reply:
Good idea